yare-engine/src/main/java/com/sabre/oss/yare/engine/executor/DefaultRulesExecutor.java

```diff
@@ -45,19 +45,32 @@
 import java.util.Map.Entry;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
-public class DefaultRulesExecutor implements RulesExecutor, Wrapper, EvictableCache {
+public class DefaultRulesExecutor implements RulesExecutor, Wrapper, EvictableCache, EngineListener {
     private static final Logger log = LoggerFactory.getLogger(DefaultRulesExecutor.class);
 
     private final Map<Class<?>, String> typeNames = new ConcurrentHashMap<>();
     private final LoadingCache<String, RuntimeRules> runtimeRulesCache;
     private final ExecutorConfiguration configuration;
+    private final EngineController engineController;
 
-    public DefaultRulesExecutor(RulesRepository rulesRepository, RuntimeRulesBuilder runtimeRulesBuilder, ExecutorConfiguration configuration) {
+    private AtomicBoolean stopEvaluation = new AtomicBoolean(false);
+
+    public DefaultRulesExecutor(RulesRepository rulesRepository, RuntimeRulesBuilder runtimeRulesBuilder, ExecutorConfiguration configuration, EngineController engineController) {
         this.configuration = configuration;
         this.runtimeRulesCache = buildCachingContext(rulesRepository, runtimeRulesBuilder);
+        this.engineController = createEngineController(engineController);
+
+    }
+
+    private EngineController createEngineController(EngineController engineController) {
```

Comment:
It is rather `initializeEngineController()` method :)

Reply:
You are right :)